stats summary refactoring

SamPosh opened the pull request:
Added get_stats_summary and get_percentile_stats_summary . This will return the stats as string , So that the string can be used in other places. I am using this to record it in Reportportal.
currently print_stats and print_percentile_stats functions are directly record the result into console_logger. But these functions return the same results into string so that it can be used by other log handler

cyberw commented:
Did you end up just duplicating the code and not using the new functions from anywhere else? This doesnt make sense.

And `current` parameter is never used? 

Also, you need to fix flake8 & black formatting errors.

SamPosh commented:
> Did you end up just duplicating the code and not using the new functions from anywhere else? This doesnt make sense.
> 
> And `current` parameter is never used?
> 
> Also, you need to fix flake8 & black formatting errors.

print_stats_summary is directly printing to console_logger. So I refactored that function into get_stats_summary and called it from print_stats but the unit test is checking the length of string. Since it is single string the length is always 1. So it is failed. May be I will try to create it as list of string then call it from print_stats so that it won't be duplicate.

For black/flake8 formatting, as of now I am doing this from vscode web editor which doesn't have formatting option.  I am trying to setup local dev environment and try to format it.

cyberw commented:
> print_stats_summary is directly printing to console_logger. So I refactored that function into get_stats_summary and called it from print_stats but the unit test is checking the length of string. Since it is single string the length is always 1. So it is failed. Due to that I duplicated the function. May be I will try to create it as list of string then call it from print_stats so that it won't be a duplicate.

Not sure what you mean, but duplication is bad :)

> For black/flake8 formatting, as of now I am doing this from vscode web editor which doesn't have formatting option. I am trying to setup local dev environment and try to format it.

Yea... tbh, that kind of sounds like you'll have a lot of work ahead of you to make this reach the quality required for merge. You might consider having this as a fork, or trying to implement it inside your own locustfile or support lib.

SamPosh reviewed (commented):
Duplication is removed

SamPosh commented:
> > print_stats_summary is directly printing to console_logger. So I refactored that function into get_stats_summary and called it from print_stats but the unit test is checking the length of string. Since it is single string the length is always 1. So it is failed. Due to that I duplicated the function. May be I will try to create it as list of string then call it from print_stats so that it won't be a duplicate.
> 
> Not sure what you mean, but duplication is bad :)
> 
> > For black/flake8 formatting, as of now I am doing this from vscode web editor which doesn't have formatting option. I am trying to setup local dev environment and try to format it.
> 
> Yea... tbh, that kind of sounds like you'll have a lot of work ahead of you to make this reach the quality required for merge. You might consider having this as a fork, or trying to implement it inside your own locustfile or support lib.

I removed the duplicate.

cyberw commented:
Looks a lot better now. Just fix the formatting and typing issues and it might be ready.

SamPosh commented:
> Looks a lot better now. Just fix the formatting and typing issues and it might be ready.

I have fixed the type hint issue.

SamPosh commented:
> Looks a lot better now. Just fix the formatting and typing issues and it might be ready.

formatting also done. type hint is fixed as well

cyberw commented on locust/stats.py:
```diff
@@ -784,25 +784,47 @@ def on_worker_report(client_id: str, data: Dict[str, Any]) -> None:
 def print_percentile_stats(stats: RequestStats) -> None:
-    console_logger.info("Response time percentiles (approximated)")
+    stat_summary = get_percentile_stats_summary(stats)
+    for summary in stat_summary:
```
skip intermediate "summary" variable

cyberw commented on locust/stats.py:
```diff
@@ -784,25 +784,47 @@ def on_worker_report(client_id: str, data: Dict[str, Any]) -> None:
 
 def print_stats(stats: RequestStats, current=True) -> None:
+    stat_summary = get_stats_summary(stats, current)
+    for summary in stat_summary:
```
skip intermediate "start_summary" variable. Also, maybe "line" is a more appropriate name than "summary"?

cyberw commented on locust/stats.py:
```diff
@@ -784,25 +784,47 @@ def on_worker_report(client_id: str, data: Dict[str, Any]) -> None:
+    Percentile stats summary will be returned as list of string
+    """
+    summary = []
+    summary.append("Response time percentiles (approximated)")
```
combine this with the line above

cyberw commented:
👍 👍
